scripts/python/phdf_diff.py

```diff
@@ -66,89 +66,116 @@ def addPath():
     #sys.path.insert(0,myPath+'/../vis/python')
     #sys.path.insert(0,myPath+'/vis/python')
 
-def compare_metadata( f0, f1, tol=1.0e-12):
+def ensure_list(x):
+    return x if isinstance(x, np.ndarray) else [x]
+
+def compare_attributes(dict0, dict1):
+    keys0 = set(dict0.keys())
+    keys1 = set(dict1.keys())
+    union = keys0.union(keys1)
+    intersect = keys0.intersection(keys1)
+
+    # keys that only show up in one set
+    diff_keys = [k for k in union if (k not in keys0) or (k not in keys1)]
```

Comment:
This is nicely written!

Reply:
How come you don't just search to see if the keys are not in the intersection won't that give you the keys that are not in both?

Reply:
Or you could just use this https://www.programiz.com/python-programming/methods/set/symmetric_difference

Reply:
Oh cool, I didn't know about `symmetric_difference`, will use it